api/views.py

```diff
@@ -0,0 +1,148 @@
+import datetime
+from datetime import timedelta, datetime
+
+from django.contrib.auth.models import User, Group
+from django.db.models import Sum
+from django.shortcuts import get_object_or_404
+from rest_framework import viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+from api.serializers import UserSerializer, GroupSerializer
+from .models import Employee, Income, Outcome, Event
+from .serializers import EmployeeSerializer, EventMiniSerializer, IncomeSerializer, OutcomeSerializer, EventSerializer
+
+
+class UserViewSet(viewsets.ModelViewSet):
+    queryset = User.objects.all().order_by('-id')
+    serializer_class = UserSerializer
+
+
+class GroupViewSet(viewsets.ModelViewSet):
+    queryset = Group.objects.all()
+    serializer_class = GroupSerializer
+
+
+class EmployeeViewSet(viewsets.ModelViewSet):
+    queryset = Employee.objects.all()
+    serializer_class = EmployeeSerializer
+
+    def list(self, request, *args, **kwargs):
+        number_of_employees = request.GET.get('number_of_employees')
+
+        if number_of_employees is not None:
+            employees = Employee.objects.order_by('-salary')[:int(number_of_employees)]
+        else:
+            employees = Employee.objects.order_by('-salary')
+
+        serializer = self.get_serializer(employees, many=True)
+        return Response(serializer.data)
+
+    @action(detail=True, methods=['get'])
+    def fire(self, request, *args, **kwargs):
+        employee = self.get_object()
+        employee.hired = False
+        employee.save()
+        serializer = EmployeeSerializer(employee, many=False)
+        return Response(serializer.data)
+
+    @action(detail=True, methods=['get'])
+    def hire_again(self, request, *args, **kwargs):
+        employee = self.get_object()
+        employee.hired = True
+        employee.save()
+        serializer = EmployeeSerializer(employee, many=False)
+        return Response(serializer.data)
+
+    @action(detail=True, methods=['post'])
+    def change_position(self, request, *args, **kwargs):
+        pesel = request.POST.get('pesel')
+        new_position = request.POST.get('new_position')
+
+        if pesel is not None and new_position is not None:
+            employee = get_object_or_404(Employee, pesel=pesel)
+            employee.position = new_position
+            employee.save()
+            serializer = EmployeeSerializer(employee, many=False)
+            return Response(serializer.data)
+
+    @action(detail=False, methods=['get'])
+    def summary_cost_of_salaries(self, request, *args, **kwargs):
+        sum_of_salaries = Employee.objects.all().aggregate(Sum('salary'))
```

Comment:
Just a note, that `.all()` is unnecessary here AFAIR.

Reply:
done